### ISSUE TITLE
UK Mater List updates not propagated to verification rules and other species lists

### ISSUE THREAD

**Issue by JimBacon** [Indicia-Team/warehouse#414]
If there is a name change for a taxon in the UK Master List, resulting in a new preferred TVK (external_key), then auto-verification runs in to problems.

- All the verification rules work by matching a TVK in the rule to the external_key of the record being checked. The rules are not updated with the UK Master List so a rule that previously applied to a taxon is no longer matched. It will therefore not be flagged if the record is outside range or phenology limits.
- The identification difficulty rule will also not find a match for the updated taxon so auto-acceptance may depend upon the setting of `auto_accept_occurrences_with_null_id_difficulty`.
- Occurrences for an updated taxa, recorded against a different species list which has not been updated with the new external key, will have null cache_occurrence_functional.taxon_path. The taxon_path is obtained from the cache_taxon_paths which only holds paths for the preferred TVKs.

**Comment by JimBacon** [comment]
Hi @johnvanbreda, do you have any thoughts, for or against, 

1.  Updating the TVK of verification rules on changes to preferred TVKs in the master list,
2.  Updating the external_key of taxa in other lists when a preferred TVK changes in the master list?


The aspiration would be for verification rules and auto acceptance to continue functioning after changes to the master list.

**Comment by johnvanbreda** [comment]
@JimBacon I agree with these suggestions and I do something similar for associated taxa in the Pantheon project. In the longer term, the organism_key provides a more robust way of defining a single taxonomic concept that doesn't change quite as frequently as the accepted name TVK. This option wasn't available when these tables were designed but would be the way to go now.

**Comment by johnvanbreda** [comment]
Agreed - actually I think this is addressed (as the proposal to switch to organism key is covered by the web service). Happy to close?
